http/http.go

```diff
@@ -66,6 +120,16 @@ func PostHelper(ctx context.Context, httpClient *http.Client, stats *statsd.Clie
 
 	req, err := http.NewRequest(http.MethodPost, endpoint, &bodyBuffer)
 
+	hct := NewHTTPClientTracer(stats, action)
```

Comment:
@cory-stripe I'm confused by how `hct` sets its fields, like `DNSStart` and `DNSDone`. Based on how I'm reading the code, it looks like `hct` is being "instantiated" and then its fields are immediately referenced in creating `httpTracer`, so I'm not really sure what it's measuring.

Reply:
Ahh, I didn't realize that the `httpTracer` parameters were taking in methods, not timestamps. I see the call to the request later! Never mind!